porcupine/plugins/filemanager.py

```diff
@@ -352,18 +381,29 @@ def new_file_here(path: Path) -> None:
     if not path.is_dir():
         path = path.parent
 
-    name = ask_file_name(path, "", mode=FilenameMode.NEW)
-    if name:
-        name.touch()
+    file_name = ask_file_name(path, "", mode=FilenameMode.NEW_FILE)
+    if file_name:
+        file_name.touch()
+        get_tab_manager().event_generate("<<FileSystemChanged>>")
+        get_tab_manager().open_file(file_name)
+
+
+def new_directory_here(path: Path) -> None:
+    if not path.is_dir():
+        path=path.parent
+
+    dir_name = ask_file_name(path, "", mode=FilenameMode.NEW_DIRECTORY)
+    if dir_name :
+        dir_name.mkdir()
         get_tab_manager().event_generate("<<FileSystemChanged>>")
-        get_tab_manager().open_file(name)
 
 
 commands = [
     # Doing something to an entire project is more difficult than you would think.
     # For example, if the project is renamed, venv locations don't update.
     # TODO: update venv locations when the venv is renamed
     Command("New file here", "<<FileManager:New file>>", (lambda p: True), new_file_here),
+    Command("New directory here", "<<FileManager:New directory>>", (lambda p: True), new_directory_here), 
```

Comment:
If you right-click a file, there should be no "New directory here" option.
```suggestion
    Command("New directory here", "<<FileManager:New directory>>", (lambda p: p.is_dir()), new_directory_here), 
```

Reply:
Nevermind. I forgot about the whole `path=path.parent` thing.